cmd/dockerd/daemon_windows.go

```diff
@@ -94,6 +97,19 @@ func newCgroupParent(config *config.Config) string {
 }
 
 func (cli *DaemonCli) initContainerD(_ context.Context) (func(time.Duration) error, error) {
-	system.InitContainerdRuntime(cli.Config.Experimental, cli.Config.ContainerdAddr)
+	system.InitContainerdRuntime(cli.Config.ContainerdAddr)
+	if system.ContainerdRuntimeSupported() && cli.Config.ContainerdAddr == "" {
```

Comment:
Those Linux issues look more like a race between system containerd and Docker-started containerd, but I only skimmed them.

I'd personally think it's cleaner if, since containerd exposes a default address, we try it if we want to use containerd (new-enough OS, env-var set, random fair dice rolled a 4, etc) but don't have a user-provided socket to connect on.

That carries my assumption that we want to move away from (or at least disprefer) "standalone"/"built-in runtime" on both Linux and Windows, though, and I may be over-reaching on that. That assumption suggests to me that we don't want to accidentally default users into standalone mode due to a configuration issue or even _forgetting_ a command-line when updating a config file/service definition.

That said, it's also a packaging question: Who's distributing/installing containerd for Windows, for Docker or otherwise? I actually haven't looked at the Linux packaging for Docker for ages (it just works!) but I would guess that `docker-ce` packages depend on system-provided containerd packages these days, but obviously Docker Desktop for Windows bundles containerd in the installer, since it bundles a whole OS image.

So maybe aligning Linux and Windows is not that valuable when the rest of the packaging ecosystem is not aligned the same way, and will bring more friction than... lubrication? (Smoothosity? non-frictionalityness?)

Reply:
Static binary packages for Linux on https://download.docker.com/linux/static/stable/x86_64/ contains containerd. Docker EE packages for Windows Server are listed on https://dockermsft.azureedge.net/dockercontainer/DockerMsftIndex.json and they looks very similar so I think that it make sense that Mirantis will include containerd to those (and Docker will do same for Docker Desktop packages).